homeassistant/components/nuki/__init__.py

```diff
@@ -46,8 +63,112 @@ async def async_setup(hass, config):
 
 async def async_setup_entry(hass, entry):
     """Set up the Nuki entry."""
-    hass.async_create_task(
-        hass.config_entries.async_forward_entry_setup(entry, LOCK_DOMAIN)
+
+    hass.data.setdefault(DOMAIN, {})
+
+    try:
+        bridge = await hass.async_add_executor_job(
+            NukiBridge,
+            entry.data[CONF_HOST],
+            entry.data[CONF_TOKEN],
+            entry.data[CONF_PORT],
+            True,
+            DEFAULT_TIMEOUT,
+        )
+
+        locks, openers = await hass.async_add_executor_job(_get_bridge_devices, bridge)
+    except InvalidCredentialsException as err:
+        raise InvalidAuth from err
+    except RequestException as err:
+        raise CannotConnect from err
+
+    async def async_update_data():
+        """Fetch data from Nuki bridge."""
+        try:
+            # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+            # handled by the data update coordinator.
+            async with async_timeout.timeout(10):
+                for device in (
+                    hass.data[DOMAIN][entry.entry_id][DATA_LOCKS]
+                    + hass.data[DOMAIN][entry.entry_id][DATA_OPENERS]
+                ):
+                    for level in (False, True):
+                        try:
+                            await hass.async_add_executor_job(device.update, level)
```

Comment:
Make a function that does the update for all devices for both levels and then schedule that function once on the exexutor. Switching context is expensive.